linux-cachyos/PKGBUILD

```diff
@@ -16,11 +16,10 @@ _cachy_config=${_cachy_config-y}
 # 'bore' - select 'Burst-Oriented Response Enhancer'
 # 'bmq' - select 'BMQ Scheduler'
 # 'hardened' - select 'BORE Scheduler hardened' ## kernel with hardened config and hardening patches with the bore scheduler
-# 'cachyos' - select 'Sched-Ext Scheduler Framework Variant Scheduler with BORE Scheduler'
+# 'cachyos' - select 'CachyOS Default Scheduler (BORE)'
 # 'eevdf' - select 'EEVDF Scheduler'
```

Comment:
I'm not sure if we should keep linux-cachyos-eevdf, since it's now essentially EEVDF+sched-ext out of the box.

Reply:
but we dropped linux-cachyos-sched-ext. There's no other kernel that provides using the default EEVDF scheduler as far as I'm aware.

Reply:
> There's no other kernel that provides using the default EEVDF scheduler as far as I'm aware.

Well, linux-cachyos-server is basically the same as linux-cachyos-eevdf, except for disabling the CachyOS config options.

Reply:
> > There's no other kernel that provides using the default EEVDF scheduler as far as I'm aware.
> 
> Well, linux-cachyos-server is basically the same as linux-cachyos-eevdf, except for disabling the CachyOS config options.

That's not quite true. Aside from having default EEVDF time slices, the server kernel also has a tickrate frequency of 300Hz compared 1000Hz (not ideal for desktop). It is also not a preemptible kernel (pete is currently considering preempt=lazy for server, I don't really like it tho).

Reply:
> > > There's no other kernel that provides using the default EEVDF scheduler as far as I'm aware.
> > 
> > 
> > Well, linux-cachyos-server is basically the same as linux-cachyos-eevdf, except for disabling the CachyOS config options.
> 
> That's not quite true. Aside from having default EEVDF time slices, the server kernel also has a tickrate frequency of 300Hz compared 1000Hz (not ideal for desktop). It is also not a preemptible kernel (pete is currently considering preempt=lazy for server, I don't really like it tho).

By that I meant “disabling the CachyOS config options”. It's just close to the kernel defaults.

Reply:
> It is also not a preemptible kernel (pete is currently considering preempt=lazy for server, I don't really like it tho).

The thing is, with PREEMPT_DYNAMIC now, any kernel can be non-preemptable simply by specifying preempt=none in kernel parameters.

Reply:
While that is true, the choice of preempting RCUs or not is decided at compile time and cannot be dynamically changed, see https://lore.kernel.org/lkml/20241106201758.428310-1-ankur.a.arora@oracle.com/T/#m678eb692a0e27e8500a01698e3c3dc82dbbee499. I'm not too sure how significant this is but it is a difference to keep in mind.

Reply:
> While that is true, the choice of preempting RCUs or not is decided at compile time and cannot be dynamically changed, see https://lore.kernel.org/lkml/20241106201758.428310-1-ankur.a.arora@oracle.com/T/#m678eb692a0e27e8500a01698e3c3dc82dbbee499. I'm not too sure how significant this is but it is a difference to keep in mind.

Well, it makes sense. It's just that in my opinion we have quite a lot of kernels, which makes maintenance difficult and inconsistent between PKGBUILD, as some of them can't be supported on the latest version of kernel, such as it was many times with linux-cachyos-hardened and linux-cachyos-pds (when it was still there), while at the same time most of the users just use linux-cachyos. Like, we now don't have much difference between linux-cachyos-bore and linux-cachyos, since sched-ext is now in both of them. 

Ideally we could just have one PKGBUILD and a system of applying environment variables that would allow build derivatives like linux-cachyos-bmq, linux-cachyos-server and etc, but I haven't figured out how to best design that yet.